src/playback/view/playbacktoolbarmodel.cpp

```diff
@@ -107,7 +107,13 @@ void PlaybackToolBarModel::updateActions()
             //! NOTE In this case, we want to see the actions' description instead of the title
             settingsItems << makeMenuItem(item.action);
         } else {
-            result << makeMenuItem(item.action);
+            MenuItem* menuItem = makeMenuItem(item.action);
+            if (item.action == PLAY_ACTION_CODE && isPlayAllowed() && playbackController()->isPlaying()) {
```

Comment:
The only problem here is the duplication of logic. The same code is in PlaybackToolBarModel::onActionsStateChanges

I suggest something like this

```
MenuItem* menuItem = makeMenuItem(item.action);
if (item.action == PLAY_ACTION_CODE) {
    menuItem->setAction(playAction());
}

result << menuItem;
```

```
UiAction PlaybackToolBarModel::playAction() const
{
    UiAction action = uiActionsRegister()->action(PLAY_ACTION_CODE);

    bool isPlaying = playbackController()->isPlaying();
    action.iconCode = isPlaying ? IconCode::Code::PAUSE : IconCode::Code::PLAY;

    return action;
}
```

Also, apply the same approach to PlaybackToolBarModel::onActionsStateChanges

Reply:
That sounds good! Will resolve on Friday when I have access to a computer once more

Reply:
Should be resolved